tests/framework/web/AssetBundleTest.php

```diff
@@ -126,12 +141,36 @@ public function testSourcesPublish_AssetBeforeCopy()
         ];
         $bundle->publish($am);
 
-        $this->assertTrue(is_dir($bundle->basePath));
+        $this->assertFalse(is_dir($bundle->basePath));
         foreach ($bundle->js as $filename) {
             $publishedFile = $bundle->basePath . DIRECTORY_SEPARATOR . $filename;
             $this->assertFileNotExists($publishedFile);
         }
-        $this->assertTrue(rmdir($bundle->basePath));
+    }
+
+    public function testSourcesPublish_publishOptions_Only()
+    {
+        $view = $this->getView();
+        $am = $view->assetManager;
+
+        $bundle = new TestSourceAsset([
+            'publishOptions' => [
+                'only' => [
+                    'js/*'
+                ]
+            ],
+        ]);
+        $bundle->publish($am);
+
+        $notNeededFilesDir = $bundle->basePath . DIRECTORY_SEPARATOR . 'css';
```

Comment:
looking at the code above, this reference to CSS feels a bit magical and error prone. This should be tied to $bundle->css[0]